### PR TITLE
generic helper functions to forward bus calls from Manager to Unit object

### PR DESCRIPTION
**Pull request by poettering**
just some refactoring, no change in behaviour

### CODE REVIEW

File: src/core/dbus-manager.c [hunk 1]
```diff
@@ -617,43 +622,11 @@ static int method_start_unit_replace(sd_bus_message *message, void *userdata, sd
...
-        int r;
-
-        assert(message);
-        assert(m);
-
-        r = sd_bus_message_read(message, "s", &name);
-        if (r < 0)
-                return r;
-
-        r = bus_get_unit_by_name(m, message, name, &u, error);
```
Comment (yuwata) [inline src/core/dbus-manager.c:614]: `method_generic_unit_operation()` uses `manager_load_unit()` instead of `bus_get_unit_by_name()`. Is this change intended? Why not use `bus_get_unit_by_name()` in `method_generic_unit_operation()`?

File: src/core/dbus-manager.c [hunk 2]
```diff
@@ -629,76 +629,42 @@ static int method_reset_failed_unit(sd_bus_message *message, void *userdata, sd_
         return method_generic_unit_operation(message, userdata, error, bus_unit_method_reset_failed);
 }
 
-static int method_set_unit_properties(sd_bus_message *message, void *userdata, sd_bus_error *error) {
-        Manager *m = userdata;
+static int method_generic_unit_operation_loaded_only(sd_bus_message *message, Manager *m, sd_bus_error *error, sd_bus_message_handler_t handler) {
```
Comment (yuwata) [inline src/core/dbus-manager.c]: Maybe, instead of creating new function, how about adding a boolean argument `validate_load_state`?

File: src/core/dbus-manager.c [hunk 3]
```diff
@@ -629,76 +629,42 @@ static int method_reset_failed_unit(sd_bus_message *message, void *userdata, sd_
         return method_generic_unit_operation(message, userdata, error, bus_unit_method_reset_failed);
```
Comment (yuwata) [inline src/core/dbus-manager.c]: Not a matter, but there exists a typo in commit message: 'agfain' -> 'gain'.

**Comment by yuwata** [comment]
I am not sure it is related, but semaphore CI fails with the following:
```
======================================================================
FAIL: test_no_failed (__main__.ServicesTest)
No failed units
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/tmp/autopkgtest-lxc.8nwjw470/downtmp/build.gjQ/src/debian/tests/boot-and-services", line 67, in test_no_failed
    self.assertEqual(failed, [])
AssertionError: Lists differ: ['NetworkManager-wait-online.service loade[40 chars]ine'] != []

First list contains 1 additional elements.
First extra element 0:
'NetworkManager-wait-online.service loaded failed failed Network Manager Wait Online'

+ []
- ['NetworkManager-wait-online.service loaded failed failed Network Manager Wait '
-  'Online']

----------------------------------------------------------------------
Ran 23 tests in 5.308s

FAILED (failures=1, skipped=7)
autopkgtest [18:50:57]: test boot-and-services: -----------------------]
autopkgtest [18:50:58]: test boot-and-services:  - - - - - - - - - - results - - - - - - - - - -
boot-and-services    FAIL non-zero exit status 1
```

**Comment by poettering** [comment]
@yuwata is completely right, the old version was pretty borked...

I force pushed a new version, completely rewritten. It now correctly uses _get_unit and _load_unit at the right places again, and merges the two calls into a single call taking a flags parameter.

This should be more correct. PTAL!

File: src/core/dbus-manager.c [hunk 4]
```diff
@@ -785,43 +719,11 @@ static int method_list_units_by_names(sd_bus_message *message, void *userdata, s
...
-        r = sd_bus_message_read(message, "s", &name);
-        if (r < 0)
-                return r;
-
-        r = bus_get_unit_by_name(m, message, name, &u, error);
-        if (r < 0)
-                return r;
-
-        return bus_unit_method_attach_processes(message, u, error);
+        return method_generic_unit_operation(message, userdata, error, bus_unit_method_attach_processes, GENERIC_UNIT_VALIDATE_LOADED);
```
Comment (keszybz) [inline src/core/dbus-manager.c:740]: Why does this one get validation? Please add a comment.

**Comment by poettering** [comment]
force pushed a new version now, which adds comments to all calls using the new generic implementation. PTAL!

**Comment by keszybz** [comment]
The comments are very helpful, thanks. LGTM.
